include/picongpu/fields/MaxwellSolver/Yee/YeePML.kernel

```diff
@@ -0,0 +1,252 @@
+                    /// ey( index ) := splitFieldE.eyx( index ) + splitFieldE.eyz( index )
+                    /// ez( index ) := splitFieldE.ezx( index ) + splitFieldE.ezy( index )
+                }
+            );
+        }
+    };
+
+    /** compute magnetic field
+     *
+     * @tparam T_numWorkers number of workers
+     * @tparam T_BlockDescription field (electric and magnetic) domain description
+     */
+    template<
+        uint32_t T_workers,
+        typename T_BlockDescription
+    >
+    struct KernelUpdateBHalf
+    {
+        /** update magnetic field
+         *
+         * @tparam T_Curl curl functor type
+         * @tparam T_EBox pmacc::DataBox, electric field box type
+         * @tparam T_BBox pmacc::DataBox, magnetic field box type
+         * @tparam T_SplitBBox PML split magnetic field box type
+         * @tparam T_Mapping mapper functor type
+         * @tparam T_Acc alpaka accelerator type
+         *
+         * @param acc alpaka accelerator
+         * @param curl functor to calculate the electric field, interface must be
+         *             `operator()(T_EBox)`
+         * @param fieldB magnetic field iterator
+         * @param fieldE electric field iterator
+         * @param mapper functor to map a block to a supercell
+         */
+        template<
+            typename T_Curl,
+            typename T_EBox,
+            typename T_BBox,
+            typename T_SplitBBox,
+            typename T_Mapping,
+            typename T_Acc
+        >
+        DINLINE void operator()(
+            T_Acc const & acc,
+            T_Curl const curl,
+            T_BBox fieldB,
+            T_SplitBBox splitFieldB,
+            T_EBox const fieldE,
+            T_Mapping mapper
+        ) const
+        {
+            using namespace mappings::threads;
+
+            constexpr uint32_t cellsPerSuperCell = pmacc::math::CT::volume< SuperCellSize >::type::value;
+            constexpr uint32_t numWorkers = T_workers;
+
+            uint32_t const workerIdx = threadIdx.x;
+
+            auto cachedE = CachedBox::create<
+                0u,
+                typename T_EBox::ValueType
+            >(
+                acc,
+                T_BlockDescription( )
+            );
+
+            nvidia::functors::Assign assign;
+            DataSpace< simDim > const block( mapper.getSuperCellIndex( DataSpace< simDim >( blockIdx ) ) );
+            DataSpace< simDim > const blockCell = block * MappingDesc::SuperCellSize::toRT( );
+
+            auto fieldEBlock = fieldE.shift( blockCell );
+
+            ThreadCollective<
+                T_BlockDescription,
+                numWorkers
+            > collective( workerIdx );
+
+            collective(
+                acc,
+                assign,
+                cachedE,
+                fieldEBlock
+            );
+
+            __syncthreads();
+
+            constexpr float_X dt = DELTA_T;
```

Comment:
Do we need to change this to `constexpr float_X dt = DELTA_T / 2;` since we advance the B filed only for a half time step.